### PR TITLE
Remove historic to do comments from libaom code

### PR DESCRIPTION
**Pull request by hassount**

### CODE REVIEW

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 1]
```diff
@@ -1288,9 +1283,7 @@ typedef enum ATTRIBUTE_PACKED {
 
 // 4 scratch frames for the new frames to support a maximum of 4 cores decoding
 // in parallel, 3 for scaled references on the encoder.
-// TODO(hkuang): Add ondemand frame buffers instead of hardcoding the number
 // of framebuffers.
```
Comment (tianjunwork) [inline Source/Lib/Common/Codec/EbDefinitions.h:1286]: line 1286 and 1287 should also be removed together

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 2]
```diff
@@ -1317,7 +1310,6 @@ MAX_NUM_TEMPORAL_LAYERS * MAX_NUM_SPATIAL_LAYERS
 static INLINE int32_t is_valid_seq_level_idx(uint8_t seq_level_idx) {
     return seq_level_idx < 24 || seq_level_idx == 31;
 }
-// TODO(jingning): Turning this on to set up transform coefficient
 // processing timer.
```
Comment (tianjunwork) [inline Source/Lib/Common/Codec/EbDefinitions.h:1313]: line 1313 should be removed.

File: Source/Lib/Common/Codec/EbRestoration.h [hunk 3]
```diff
@@ -84,7 +84,6 @@ void eb_apply_selfguided_restoration_c(const uint8_t *dat8, int32_t width, int32
 #define RESTORATION_UNITPELS_MAX (RESTORATION_UNITPELS_HORZ_MAX * RESTORATION_UNITPELS_VERT_MAX)
 
 // Two 32-bit buffers needed for the restored versions from two filters
-// TODO(debargha, rupert): Refactor to not need the large tilesize to be stored
 // on the decoder side.
```
Comment (tianjunwork) [inline Source/Lib/Common/Codec/EbRestoration.h:87]: line 87 should be removed.

File: Source/Lib/Common/ASM_SSSE3/av1_inv_txfm_ssse3.c [hunk 4]
```diff
@@ -1643,7 +1641,6 @@ static void iadst4_new_sse2(const __m128i *input, __m128i *output, int8_t cos_bi
     }
 }
 
-// TODO(binpengsmail@gmail.com):
 // To explore the reuse of VP9 versions of corresponding SSE2 functions and
 // evaluate whether there is a possibility for further speedup.
```
Comment (tianjunwork) [inline Source/Lib/Common/ASM_SSSE3/av1_inv_txfm_ssse3.c:1645]: line 1644 and 1645 are still needed?

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 5]
```diff
@@ -367,7 +367,6 @@ one more than the minimum. */
 #define DIST_PRECISION_BITS 4
 #define DIST_PRECISION (1 << DIST_PRECISION_BITS) // 16
 
-// TODO(chengchen): Temporal flag serve as experimental flag for WIP
 // bitmask construction.
 // Shall be removed when bitmask code is completely checkedin
 #define LOOP_FILTER_BITMASK 0
```
Comment (tianjunwork) [inline Source/Lib/Common/Codec/EbDefinitions.h:372]: line 370 and 371 are still needed?

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 6]
```diff
@@ -873,7 +871,6 @@ typedef enum ATTRIBUTE_PACKED {
     ADST_1D,
     FLIPADST_1D,
     IDTX_1D,
-    // TODO(sarahparker) need to eventually put something here for the
     // mrc experiment to make this work with the ext-tx pruning functions
```
Comment (tianjunwork) [inline Source/Lib/Common/Codec/EbDefinitions.h:874]: line 874 should be removed

File: Source/Lib/Common/Codec/convolve.h [hunk 7]
```diff
@@ -61,7 +61,6 @@ static INLINE ConvolveParams get_conv_params_no_round(int32_t ref, int32_t do_av
         conv_params.round_0 += intbufrange - 16;
         if (!is_compound) conv_params.round_1 -= intbufrange - 16;
     }
-    // TODO(yunqing): The following dst should only be valid while
     // is_compound = 1;
```
Comment (tianjunwork) [inline Source/Lib/Common/Codec/convolve.h:64]: line 64 should be removed.

File: Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.c [hunk 8]
```diff
@@ -827,7 +825,6 @@ void setup_ref_mv_list(PictureControlSet *pcs_ptr, const Av1Common *cm, const Ma
     //CHKN finish the Tables.
 
     if (rf[1] > NONE_FRAME) {
-        // TODO(jingning, yunqing): Refactor and consolidate the compound and
         // single reference frame modes. Reduce unnecessary redundancy.
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.c:828]: line 828 should be removed

File: Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.c [hunk 9]
```diff
@@ -985,7 +981,6 @@ void setup_ref_mv_list(PictureControlSet *pcs_ptr, const Av1Common *cm, const Ma
                     if (stack_idx == refmv_count[ref_frame]) {
                         ref_mv_stack[ref_frame][stack_idx].this_mv = this_mv;
 
-                        // TODO(jingning): Set an arbitrary small number here. The weight
                         // doesn't matter as long as it is properly initialized.
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.c:984]: line 984

**Comment by 1480c1** [comment]
Found some more
https://github.com/OpenVisualCloud/SVT-AV1/blob/ff3a177bc62e9a5e6f8ad22ea7450b173a39f42e/Source/Lib/Common/ASM_AVX2/EbIntraPrediction_Intrinsic_AVX2.c#L405-L406

https://github.com/OpenVisualCloud/SVT-AV1/blob/bad003098cbedf95d0af5b06678aedd1346e03f6/Source/Lib/Encoder/Codec/EbPsnr.c#L27-L29

I don't know how relevant these are though

**Comment by tianjunwork** [comment]
Thank you @1480c1 Chris! They are removed in new commit.

**Pull request merged**